### ISSUE TITLE
上次我弄得pb文件你有帮我试试吗

### ISSUE THREAD

**Issue by chendaming** [jakajacky/tensorFlowDemo#1]
链接: https://pan.baidu.com/s/1bpcZE6F 密码: fmmg     

![default](https://user-images.githubusercontent.com/9350899/30257146-7839805c-96e2-11e7-9bd2-a851df28bfa5.png)

Running model failed:Not found: FeedInputs: unable to find feed output Mul
如何知道有没有input_layer_name   是  Mul
 我是通过下图来优化模型的  之前生成80M的模型的时候 没出现这个报错  
![default](https://user-images.githubusercontent.com/9350899/30257146-7839805c-96e2-11e7-9bd2-a851df28bfa5.png)

**Comment by jakajacky** [comment]
你在.py文件中应该能找到对应的input node，或者是placeholder，我没有你的py文件，不过打印出Nodes看了看，
<img width="486" alt="2017-09-11 3 46 03" src="https://user-images.githubusercontent.com/16397721/30263684-84a77058-9708-11e7-930d-8b4b44a1e1a7.png">
应该是这个‘input’了，你试试input_layer_name = "input"

**Comment by chendaming** [comment]
Running model failed:Invalid argument: Tried to explicitly squeeze dimension 1 but dimension was not 1: 4
	 [[Node: MobilenetV1/Logits/SpatialSqueeze = Squeeze[T=DT_FLOAT, squeeze_dims=[1, 2], _device="/job:localhost/replica:0/task:0/cpu:0"](MobilenetV1/Logits/Conv2d_1c_1x1/BiasAdd)]]

改了报这个 是我训练出问题了吗  我训练完没有.py文件
我训练的方法 
![default](https://user-images.githubusercontent.com/9350899/30264911-0d3f49be-970d-11e7-96ea-ec04390924aa.png)
   训练出来的模型 
![default](https://user-images.githubusercontent.com/9350899/30264899-00d8e7b6-970d-11e7-94f1-8c6a1be10798.png)

**Comment by jakajacky** [comment]
看训练模型的py就行。这个retyain.py里面一般就有相关的input和output的描述。这个错误，应该是你在使用模型的时候，设置的维度与模型实际的维度 不匹配。

**Comment by chendaming** [comment]
我使用的是默认的tensorflow/tensorflow/examples/image_retraining/retyain.py   但没找到有类似的设置

**Comment by jakajacky** [comment]
```
auto nodeCount = tensorflow_graph.node_size();
NSLog(@"Node count: %d", nodeCount);
for (auto i = 0; i < nodeCount; ++i) {
    auto node = tensorflow_graph.node(i);
    NSLog(@"Node %d: %s '%s'", i, node.op().c_str(), node.name().c_str());
}
```
从py里面看不出来，也可以加载完模型，打印一下node，一般placeholder对应input node  softmax对于output node。我还没见过其他的，这也跟python怎么写的有关。

add：不过我说的这个不是一定的，有些人写的py就是有些自定义的input output等等node名称，感觉得理解好训练时用的py才能找到对应的名称

**Comment by chendaming** [comment]
你看看这个https://github.com/tensorflow/tensorflow/commit/055500bbcea60513c0160d213a10a7055f079312   他里面也没说需要修改retyain.py 的参数    

难道要修改这些参数吗 ？ 
![default](https://user-images.githubusercontent.com/9350899/30356723-2c0f2448-986d-11e7-8e58-d1149eb1e3d7.png)

**Comment by chendaming** [comment]
谢谢啊   原来之前我一直都看错了retyain.py  那个1.0的 ····     
https://github.com/JieHe96/iOS_Tensorflow_ObjectDetection_Example 这个demo 你能运行吗 我怎么一直报tensorflow/tensorflow/contrib/makefile/downloads/protobuf 里的版本不对 一个需要3.3 一直需要3.1 版本 。 不知道怎么处理

**Comment by jakajacky** [comment]
运行的时候还是编译的时候，TF报错还是demo里面的代码报错？我看JieHe的代码更新了，里面加了一些 #error，限制了protobuf的版本

**Comment by chendaming** [comment]
编译的时候 这个版本  
![default](https://user-images.githubusercontent.com/9350899/30364440-d4d51cd0-9896-11e7-82fb-6552a7b11bf1.png)

protobuf 的版本是build的时候就固定了 我试过下载后放入3.3的 进去 但编译.a等文件就失败。

**Comment by jakajacky** [comment]
我下载最新的也是这个问题，我觉得情况是这样的，这个版本的demo，要求最新的TensorFlow1.4，所以，需要重新编译新的TF了。我看JieHe的readme里教程是这么写的，需要重新编译最新的1.4，是不是你现在用的是老版本？

**Comment by chendaming** [comment]
对的  我使用的是旧版本 那我下载1.4的试试

**Comment by jakajacky** [comment]
ok下载1.4，应该要从新来一遍，JieHe的教程里也做了更新，参考一下

**Comment by chendaming** [comment]
对了 我想问一个问题   之前我训练的 TensorFlow 的 imagenet 模型能获得 具体是别的物体的再图片的位置吗?  这是不是需要别的模型才能做到的？

**Comment by jakajacky** [comment]
从我目前看懂的代码里，觉得目前是没有关于‘位置’的输出的，不过我觉得应该有对于位置的计算，毕竟识别也要分区域，至于需要做哪些工作，将‘位置’信息暴露，然后作为输出，我是写不出来的，对于python两眼一抹黑^_^

**Comment by chendaming** [comment]
恩   之前看coreML会生成模型对应的方法 估计可以用来调用 但直接用模型的话 tensorflow_utils 会不会这个文件会根据不同模型来更改的 但目前不知道如何生成这个文件
